src/components/search/detailed/DetailedSearchResults.js

```diff
@@ -0,0 +1,243 @@
+        setSelectedTab(selectedTab);
+    };
+    const dataTabId = build(baseId, ids.DATA_SEARCH_RESULTS_TAB);
+    const appsTabId = build(baseId, ids.APPS_SEARCH_RESULTS_TAB);
+    const analysesTabId = build(baseId, ids.ANALYSES_SEARCH_RESULTS_TAB);
+
+    useEffect(() => {
+        if (filter === searchConstants.ALL || filter === searchConstants.DATA) {
+            setSelectedTab(TABS.data);
+        } else if (filter === searchConstants.APPS) {
+            setSelectedTab(TABS.apps);
+        } else if (filter === searchConstants.ANALYSES) {
+            setSelectedTab(TABS.analyses);
+        }
+    }, [filter, setSelectedTab]);
+
+    const dataTabIcon =
+        selectedTab === TABS.data ? "/data_selected.png" : "/icon-data.png";
+    const appsTabIcon =
+        selectedTab === TABS.apps ? "/apps_selected.png" : "/icon-apps.png";
+    const analysesTabIcon =
+        selectedTab === TABS.analyses
+            ? "/analyses_selected.png"
+            : "/icon-analyses.png";
+
+    const dataTab = (
+        <Tab
+            value={TABS.data}
+            key={dataTabId}
+            id={dataTabId}
+            label={
+                isMobile ? dataCount : t("dataSearchTab", { count: dataCount })
+            }
+            classes={{ selected: classes.tabSelected }}
+            icon={
+                <img
+                    src={`${dataTabIcon}`}
+                    alt={t("data")}
+                    className={classes.tabIcon}
+                />
+            }
+        />
+    );
+
+    const appsTab = (
+        <Tab
+            value={TABS.apps}
+            key={appsTabId}
+            id={appsTabId}
+            label={
+                isMobile ? appsCount : t("appsSearchTab", { count: appsCount })
+            }
+            classes={{ selected: classes.tabSelected }}
+            icon={
+                <img
+                    src={`${appsTabIcon}`}
+                    alt={t("apps")}
+                    className={classes.tabIcon}
+                />
+            }
+        />
+    );
+
+    const analysesTab = (
+        <Tab
+            value={TABS.analyses}
+            key={analysesTabId}
+            id={analysesTabId}
+            label={
+                isMobile
+                    ? analysesCount
+                    : t("analysesSearchTab", { count: analysesCount })
+            }
+            classes={{ selected: classes.tabSelected }}
+            icon={
+                <img
+                    src={`${analysesTabIcon}`}
+                    alt={t("analyses")}
+                    className={classes.tabIcon}
+                />
+            }
+        />
+    );
+
+    let tabsToRender = [dataTab, appsTab, analysesTab];
+    if (filter === searchConstants.DATA) {
+        tabsToRender = dataTab;
+    } else if (filter === searchConstants.APPS) {
+        tabsToRender = appsTab;
+    } else if (filter === searchConstants.ANALYSES) {
+        tabsToRender = analysesTab;
+    }
```

Comment:
I was able to fix the console warnings with an unexpected solution. I added `selectedTab` to URL query param so that browser history will remember the tab the user was in. When users navigates out of search and hits `Back`, they will come back to same tab they were previously in. As a bonus, this fixed the warnings too. This is in #166 .